integration-tests/sample_test.go

```diff
@@ -58,3 +62,156 @@ func TestAuth(t *testing.T) {
 		t.Fatal("UserID  mismatch")
 	}
 }
+
+func TestValidateSignature(t *testing.T) {
+	if options.proxyHost == "" || options.proxyPort == "" || options.sampleName == "" {
+		t.Skip("sample-name, proxy-host or proxy-port are missing => skip TestValidateSignature")
+	}
+
+	// All samples generate a new state and redirect us to an /authorize URL, if we're not logged in.
+	httpClient := &http.Client{
+		CheckRedirect: func(req *http.Request, via []*http.Request) error {
+			return http.ErrUseLastResponse
+		},
+	}
+
+	client := newSampleClient(options.sampleURL, httpClient)
+	restartSample(client)
+
+	m := newModifier(modifySignatureHandler)
+	m.start()
+
+	defer m.stop()
+
+	payload := struct {
+		ModifyURL string `json:"modifyUrl"`
+	}{
+		ModifyURL: "http://127.0.0.1:" + fmt.Sprint(m.getPort()),
+	}
+
+	prreq, err := newRequest(fmt.Sprintf("http://%v:%v/session", options.proxyHost, options.proxyPort), "POST", payload)
+	if err != nil {
+		t.Fatalf("Error making request to the proxy: %v", err)
+	}
+
+	_, _, err = getResponse(prreq, httpClient)
+	if err != nil {
+		t.Fatalf("Error when receiving the proxy response: %v", err)
+	}
```

Comment:
ProxyClient may be? Too much setup logic in this test. Or even may be make it Modifiers responsibility to manage the session.